File: app/src/main/java/com/dhu/dhu_timetable/repo/UsersRepo.java
package com.dhu.dhu_timetable.repo;

import android.util.Log;

import androidx.lifecycle.MutableLiveData;

import com.dhu.dhu_timetable.model.LoginModel;
import com.dhu.dhu_timetable.service.APIService;
import com.dhu.dhu_timetable.service.RetrofitConnect;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class UsersRepo {

    // 싱글톤 패턴
    private static UsersRepo instance;

    public static UsersRepo getInstance() {
        if (instance == null)
            instance = new UsersRepo();
        return instance;
    }

    private final APIService service;
    String TAG = "janghee";

    public UsersRepo() {
        service = RetrofitConnect.INSTANCE.getRetrofitClient().create(APIService.class);
        Log.d(TAG, "UserRepo: 성공");
    }

    public MutableLiveData<LoginModel> getUser(String email) {
        MutableLiveData<LoginModel> user = new MutableLiveData<>();
        service.getUser(email).enqueue(new Callback<LoginModel>() {
            @Override
            public void onResponse(Call<LoginModel> call, Response<LoginModel> response) {
                if (response.isSuccessful()) {
                    user.setValue(response.body());
                    Log.e(TAG, "onResponse: "+response.body().getName() + response.body().getEmail());
                }
            }

            @Override
            public void onFailure(Call<LoginModel> call, Throwable t) {
                Log.d(TAG, "Repo onFailure: "+t.getMessage());
            }
        });

        return user;
    }

    public void setUser(String email, String name, String profile) {
        MutableLiveData<LoginModel> user = new MutableLiveData<>();
        service.checkUser(email, name, profile).enqueue(new Callback<LoginModel>() {
            @Override
            public void onResponse(Call<LoginModel> call, Response<LoginModel> response) {
                if (response.isSuccessful()) {
                    user.setValue(response.body());
                    Log.d(TAG, "onResponse: 유저등록");
                }
            }

            @Override
            public void onFailure(Call<LoginModel> call, Throwable t) {
                Log.d(TAG, "Repo onFailure: "+t.getMessage());
            }
        });

    }

}

File: app/src/main/java/com/dhu/dhu_timetable/repo/UsersRepo.kt
package com.dhu.dhu_timetable.repo

import androidx.lifecycle.MutableLiveData
import com.dhu.dhu_timetable.model.LoginModel
import com.dhu.dhu_timetable.network.APIService
import com.dhu.dhu_timetable.network.RetrofitConnect
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class UsersRepo {
    private val service: APIService by lazy { RetrofitConnect.getRetrofitClient()!!.create(APIService::class.java) }

    companion object {
        @Volatile private var instance: UsersRepo? = null
        @JvmStatic
        fun getInstance(): UsersRepo =
                instance ?: synchronized(this) {
                    instance ?: UsersRepo().also { instance = it }
                }
    }

    fun getUser(email: String): MutableLiveData<LoginModel> {
        val user = MutableLiveData<LoginModel>()
        service.getUser(email).enqueue(object : Callback<LoginModel> {
            override fun onResponse(call: Call<LoginModel>, response: Response<LoginModel>) {
                if (response.isSuccessful) {
                    user.value = response.body()

                }
            }

            override fun onFailure(call: Call<LoginModel>, t: Throwable) {

            }
        })
        return user
    }

    fun setUser(email: String, name: String, profile: String) {
        val user = MutableLiveData<LoginModel>()
        service.checkUser(email, name, profile).enqueue(object : Callback<LoginModel> {
            override fun onResponse(call: Call<LoginModel>, response: Response<LoginModel>) {
                if (response.isSuccessful) {
                    user.value = response.body()

                }
            }

            override fun onFailure(call: Call<LoginModel>, t: Throwable) {

            }
        })
    }
}